hazelcast-client/src/main/java/com/hazelcast/client/spi/ProxyManager.java

```diff
@@ -458,6 +464,58 @@ public Address findNextAddressToSendCreateRequest() {
     }
 
     public Collection<? extends DistributedObject> getDistributedObjects() {
+        try {
+            ClientMessage request = ClientGetDistributedObjectsCodec.encodeRequest();
+            ClientInvocationFuture future = new ClientInvocation(client, request, client.getName()).invoke();
+            ClientMessage response = future.get();
+            processDistributedObjectInfos(response);
+            return getLocalDistributedObjects();
+        } catch (Exception e) {
+            throw rethrow(e);
+        }
+    }
+
+    class SyncDistributedObjectsTask implements Runnable {
+        @Override
+        public void run() {
+            ClientMessage request = ClientGetDistributedObjectsCodec.encodeRequest();
+            ClientInvocationFuture future = new ClientInvocation(client, request, client.getName()).invoke();
+            future.andThen(new ExecutionCallback<ClientMessage>() {
+                @Override
+                public void onResponse(ClientMessage response) {
+                    processDistributedObjectInfos(response);
+                }
+
+                @Override
+                public void onFailure(Throwable t) {
+
+                }
+            });
+        }
+    }
+
+    private void processDistributedObjectInfos(ClientMessage response) {
```

Comment:
do we need to synchronize this method? would the order change the order or type of DistributedObject events delivered to local distributed object listeners?

Reply:
DistributedObject events are not fired from this method. Since eventually everything will be ok, I don't see any need for synchronisation.